internal/services/cognitive/cognitive_commitment_plan_resource_test.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type CognitiveCommitmentPlanTestResource struct{}
+
+func TestAccCognitiveCommitmentPlan_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_commitment_plan", "test")
+	r := CognitiveCommitmentPlanTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCognitiveCommitmentPlan_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_commitment_plan", "test")
+	r := CognitiveCommitmentPlanTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccCognitiveCommitmentPlan_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_commitment_plan", "test")
+	r := CognitiveCommitmentPlanTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCognitiveCommitmentPlan_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_commitment_plan", "test")
+	r := CognitiveCommitmentPlanTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r CognitiveCommitmentPlanTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := commitmentplans.ParseAccountCommitmentPlanID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Cognitive.CommitmentPlansClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r CognitiveCommitmentPlanTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```

Comment:
The `provider` definition must be put in each testcase rather than in `template`

Reply:
Fixed.